app/precompiled.py

```diff
@@ -104,17 +105,32 @@ def add_tag_to_precompiled_letter():
 @statsd(namespace="template_preview")
 def validate_pdf_document():
     encoded_string = request.get_data()
+    generate_preview_pngs = request.args.get('include_preview') in ['true', 'True', '1']
 
     if not encoded_string:
         abort(400)
 
-    file_data = BytesIO(encoded_string)
+    data = {
+        'result': validate_document(BytesIO(encoded_string)),
+    }
+
+    if not generate_preview_pngs:
+        return jsonify(data)
+
+    if not data['result']:
+        data['message'] = 'PDF contents outside printable areas'
+        pages = overlay_template_areas(BytesIO(encoded_string), overlay=True)
+
+    else:
+        data['message'] = 'OK'
```

Comment:
@karlchillmaid Hi Karl, do you have a better idea for an OK message? It will appear on a banner on the page where a user can validate if their pdf document is within printable boundaries.

We set the validation failed message to 'PDF contents outside printable areas', but we are also opened to change ☺️

Reply:
Validation fail: `Content in this PDF is outside the printable area`

I'm working on a more descriptive confirmation message than 'OK', but I'll need some more information about the user journey first.

Reply:
@CrystalPea I'd like to try `Your PDF passed the layout check` for the time being.

We can revisit it as and when this feature develops or becomes self-service.